src/psyclone/dynamo0p3.py

```diff
@@ -2709,15 +2710,19 @@ def gen_code(self, parent):
         self.evaluators.initialise_basis_fns(invoke_sub)
 
         if self.is_coloured():
-            # Add declarations of the colour map and array holding the
-            # no. of cells of each colour
-            invoke_sub.add(DeclGen(parent, datatype="integer",
+            # Declare the colour map
+            declns = ["cmap(:,:)"]
+            if not config.DISTRIBUTED_MEMORY:
+                # Declare the array holding the no. of cells of each
+                # colour
+                declns.append("ncp_colour(:)")
+            invoke_sub.add(DeclGen(invoke_sub, datatype="integer",
                                    pointer=True,
-                                   entity_decls=["cmap(:,:)",
-                                                 "ncp_colour(:)"]))
-            # Declaration of variable to hold the number of colours
-            invoke_sub.add(DeclGen(parent, datatype="integer",
-                                   entity_decls=["ncolour"]))
+                                   entity_decls=declns))
+            if not config.DISTRIBUTED_MEMORY:
```

Comment:
I did not join them as it would have resulted in a different declaration ordering which would potentially affect a number of tests. Instead I explained why there is a difference between distributed memory and not, in the two places separately.